guides/hack/11-built-in-types/74-nothing.md

```diff
@@ -15,14 +15,38 @@ Let's work out the hierarchy of scalar types. Forget about nullable types and `d
 
 The important thing to note here is that `nothing` is never between two types. `nothing` only shows up right below a type with no other subtypes.
 
-Okay that was all very academical, but "what can I use it for?", I hear you ask.
+## Usages
+
+When defining a function that will never return (it either throws, loops forever, or terminates the request) you can use `nothing` for the return type. This gives more information to the caller than `void` and is more flexible than [noreturn](./noreturn). `nothing` can be used in expressions (like `nullable T ?? nothing`) and it will typecheck "as if it wasn't there", since `(T & nothing)` is _just_ `T`.
+
+`nothing` can be used to create a `throw` expression in this way. 
+
+@@ nothing-examples/throw-as-an-expression.php @@
+
+<hr>
+
+When writing a new bit of functionality, you may write an interface which does not have any classes implementing it yet. When trying to get some work done, you may find yourself wanting to call a method which requires such an interface in its arguments. You now have a choice. You can either ignore the typeerror and continue working on the thing you were trying to do. Or you stop doing what you were doing and write a class that implements this interface to satisfy the typehint. Both are not ideal. The first option keeps a typeerror, which may be distracting for you. The second option completely breaks your flow state and when you come back, you will be less effective while you're getting back into your flow.
+
+`nothing` is an amazing solution which allows you to keep working what you were working on, without having the nagging feeling of the typeerror that needs to be fixed. The trick is defining a function with the return type of `nothing`. This function may not return anything (just like a `noreturn` function). You can now use this function as a placeholder for the value which you could not create. I have called this function (and an opaque type alias) `undefined`, as an homage to Haskell [undefined](https://wiki.haskell.org/Undefined).
```

Comment:
I don't have a strong opinion but I feel like these two paragraphs could probably be shortened to one sentence, something like "This allows you to implement a placeholder function that can be used in place of any type without causing type errors." and let people decide for themselves whether that's something they want or what they'd use it for (the example illustrates your use case pretty well I think, probably doesn't need a long text description on top of that).

Personally, I'm not sure whether using `nothing` to silence a type error like this (possibly turning it into a runtime error unless I have a staging environment like you described) is something I'd want to do, so that's another reason why it might be better to present the example more factually, not as a recommendation.

Reply:
I guess I couldn't contain my excitement about this "little hack". It is indeed very out of place when compared to the rest of the docs. I'll rephrase to be more in line with the rest of the docs and trim the excess.

This trick has brought back the ability to take some micro short term tech debt during development. Finishing my thought / thread of work and coming back to it later. I run into the given example use case a lot.